registry/txt.go

```diff
@@ -135,6 +144,23 @@ func (im *TXTRegistry) Records(ctx context.Context) ([]*endpoint.Endpoint, error
 				ep.Labels[k] = v
 			}
 		}
+		// Handle the migration of TXT records created before the new format (introduced in v0.12.0)
+		if len(txtRecordsMap) > 0 {
+			if plan.IsManagedRecord(ep.RecordType, im.managedRecordTypes) {
+				// Get desired TXT records and detect the missing ones
+				desiredTXTs := im.generateTXTRecord(ep)
+				missingDesiredTXT := []*endpoint.Endpoint{}
+				for _, desiredTXT := range desiredTXTs {
+					if _, exists := txtRecordsMap[desiredTXT.DNSName]; !exists {
+						missingDesiredTXT = append(missingDesiredTXT, desiredTXT)
+					}
+				}
+				if len(desiredTXTs) > len(missingDesiredTXT) {
```

Comment:
This guard condition is confusing. It's not possible that "missingDesiredTXT[s]" could be longer than "desiredTXTs." The former can only ever be shorter or the same length. Therefore, this condition is only false when the two slices are of the same length. What does it mean when they're of equal length? That there are no desired TXT records? That all of them are missing?

Reply:
If both slices are equal in lengths this means that the record is not managed by externaldns. Because if it wsa managed by externaldns, at least one TXT record would exist. I added additional comment to make this clearer.

However, this made we think of the records owned by different externaldns instances. I think that we should not migrate the TXT records for the other instances. I added a guarding condition for this [here](https://github.com/kubernetes-sigs/external-dns/blob/50f196c0db280de7a9b55a8f7fd6b0a607ac5472/registry/txt.go#L150).

@Raffo , @seh : what do you think about the migration of other (defferent txt-owner-id) externaldns instances records?

Reply:
I agree that we should only manipulate records that are managed by this ExternalDNS instance.

When trying to transfer record ownership between instances (orphaning and adopting), we've written a tool to revise the corresponding TXT records. It would be nice it one could invoke ExternalDNS with a command-line flag (or run it in a different mode, as a one-time tool rather than a long-running service) to rewrite the TXT records from an orphaning ExternalDNS instance to an adopting ExternalDNS instance.

Reply:
Yes, that's something we can think of, in the followup PRs.

Reply:
This all makes sense to me 👍